core/src/test/java/org/jdbi/v3/core/mapper/MapEntryMapperTest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.mapper;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collector;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.collector.JdbiCollectors;
+import org.jdbi.v3.core.generic.GenericType;
+import org.jdbi.v3.core.mapper.reflect.ConstructorMapper;
+import org.jdbi.v3.core.rule.H2DatabaseRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+
+public class MapEntryMapperTest {
+    @Rule
+    public H2DatabaseRule dbRule = new H2DatabaseRule();
+
+    private Handle h;
+
+    @Before
+    public void setUp() {
+        h = dbRule.getSharedHandle();
+    }
+
+    @Test
+    public void keyValueColumns() {
+        h.execute("create table config (key varchar, value varchar)");
+        h.prepareBatch("insert into config (key, value) values (?, ?)")
+                .add("foo", "123")
+                .add("bar", "xyz")
+                .execute();
+
+        Map<String, String> map = h.createQuery("select key, value from config")
+                .configure(MapEntryMapper.Config.class, cfg -> cfg.setKeyColumn("key").setValueColumn("value"))
+                .mapTo(new GenericType<Map.Entry<String, String>>() {})
+                .collect(intoContainer(new GenericType<Map<String, String>>() {}));
```

Comment:
@stevenschlansker Remember when I pushed back against adding `Query.collectInto(Class<T> | GenericType<T> | Type)`?

I think this test demonstrates that these would probably be desirable, and safe to add.

```java
public interface ResultBearing {
  <R> R collectInto(Class<R> containerType);
  <R> R collectInto(GenericType<R> containerType);
  Object collectInto(Type containerType);
}
```